swingset/src/main/java/com/nqadmin/swingset/SSDBComboBox.java

```diff
@@ -1542,51 +1537,66 @@ public void setSelectedStringValue(final String _value) {
 	}
 
 	/**
-	 * Sets the value stored in the component.
+	 * Sets the selected ComboBox item according to the specified mapping/key.
 	 * <p>
 	 * If called from updateSSComponent() from a RowSet change then the Component
 	 * listener should already be turned off. Otherwise we want it on so the
 	 * ultimate call to setSelectedIndex() will trigger an update the to RowSet.
 	 * <p>
-	 * The mappings ArrayList will be null until execute is called so
-	 * <p>
 	 * Currently not a bean property since there is no associated variable.
 	 *
-	 * @param _value database record primary key value to assign to combobox
+	 * @param _value database record mapping/key
 	 */
-	public void setSelectedValue(final long _value) {
+	public void setSelectedMapping(final long _value) {
 
-		// TODO consider firing a property change
-		// TODO what happens if user tries to pass null or if getSelectedValue() is null?
+		// HANDLED BY UNDERLYING INFRASTRUCTURE: TODO consider firing a property change
+		// HANDLED: TODO what if user pass null or if getSelectedValue() is null?
 
 		// 2020-08-03: Removing conditional as this could be called when consecutive records
 		// have the same value and we want to make sure to update the editor Strings
 
 		// IF MAPPINGS ARE SPECIFIED THEN LOCATE THE SEQUENTIAL INDEX AT WHICH THE
 		// SPECIFIED CODE IS STORED
 
-		try (Model.Remodel remodel = comboInfo.getRemodel()) {
+		try (Model.Remodel remodel = optionModel.getRemodel()) {
+			// TODO: if getAllowNull, might be empty, but still have 1 item.
 			if (!remodel.isEmpty()) {
```

Comment:
Need a better check for empty. Maybe `hasValidItems()` which takes nullItem into account.

Reply:
If you call getMappings() on an empty remodel it will throw an NPE? Or maybe calling indexOf() throws NPE when there are no mappings?

Reply:
`getMappings()` on empty model returns an empty list. If the list has has a nullItem, then a list of one element, `null`, is returned. Same with `getOptions()`, except you either empty list or list with one item an empty String.

 Hmm, does that seem right?

That's by examination, not running a test.

Reply:
Was just trying to determine if if '(!remodel.isEmpty()) {' is necessary, but I think it is to respond if there are no mappings. Just to confirm will we have [0-n) mappings if custom mappings are not specified? If so then we're really just worried about giving a log warning if the combo does not have any items (ignoring nullItem), thus your suggestion of hasValidItem().

This can happen in our app in practice.

Reply:
> will we have [0-n) mappings if custom mappings are not specified?

Yes. Good point. It's highly unlikely for there to be an empty list; bad app.

But we do have the situation where `getMappings()`/`getOptions()` returns a list that and initial null/empty item when `getAllowNull()` is true.

Reply:
Definitely will have this in practice. We have one where you select a parent record and it filters a SSDBComboBox of children. Sometimes there are no children.